vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -652,73 +601,95 @@ void ConnectionBoxMapLookahead::compute(const std::vector<t_segment_inf>& segmen
-    RoutingCosts all_costs;
+    VTR_ASSERT(REPRESENTATIVE_ENTRY_METHOD == SMALLEST);
+    RoutingCosts all_delay_costs;
+    RoutingCosts all_base_costs;
 
     /* run Dijkstra's algorithm for each segment type & channel type combination */
 #if defined(VPR_USE_TBB)
     tbb::mutex all_costs_mutex;
-    tbb::parallel_for_each(sample_points, [&](const SamplePoint& point) {
+    tbb::parallel_for_each(sample_regions, [&](const SampleRegion& region) {
 #else
-    for (const auto& point : sample_points) {
+    for (const auto& region : sample_regions) {
 #endif
         // holds the cost entries for a run
-        RoutingCosts costs;
+        RoutingCosts delay_costs;
+        RoutingCosts base_costs;
+        int total_path_count = 0;
+
+        for (auto& point : region.points) {
+            // statistics
+            vtr::Timer run_timer;
+            float max_delay_cost = 0.f;
+            float max_base_cost = 0.f;
+            int path_count = 0;
+            for (auto node_ind : point.nodes) {
+                {
+                    auto result = run_dijkstra<util::PQ_Entry_Delay>(node_ind, &delay_costs);
+                    max_delay_cost = std::max(max_delay_cost, result.first);
+                    path_count += result.second;
+                }
+                {
+                    auto result = run_dijkstra<util::PQ_Entry_Base_Cost>(node_ind, &base_costs);
+                    max_base_cost = std::max(max_base_cost, result.first);
+                    path_count += result.second;
+                }
+            }
+
+            if (path_count > 0) {
+                VTR_LOG("Expanded %d paths of segment type %s(%d) starting at (%d, %d) from %d segments, max_cost %e %e (%g paths/sec)\n",
+                        path_count, segment_inf[region.segment_type].name.c_str(), region.segment_type,
+                        point.location.x(), point.location.y(),
+                        (int)point.nodes.size(),
+                        max_delay_cost, max_base_cost,
+                        path_count / run_timer.elapsed_sec());
+            }
 
-        for (auto node_ind : point.samples) {
-            run_dijkstra<util::PQ_Entry_Delay>(node_ind, &costs);
-            run_dijkstra<util::PQ_Entry_Base_Cost>(node_ind, &costs);
+            /*
+             * if (path_count == 0) {
+             * for (auto node_ind : point.nodes) {
+             * VTR_LOG("Expanded node %s\n", describe_rr_node(node_ind).c_str());
+             * }
+             * }
+             */
+
+            total_path_count += path_count;
+            if (total_path_count > MIN_PATH_COUNT) {
+                break;
+            }
         }
 
 #if defined(VPR_USE_TBB)
         all_costs_mutex.lock();
 #endif
 
-        for (auto node_ind : point.samples) {
-            VTR_LOG("Expanded node %s\n", describe_rr_node(node_ind).c_str());
+        if (total_path_count == 0) {
+            VTR_LOG("No paths found for sample region %s(%d, %d)\n",
```

Comment:
Should probably be a VTR_WARNING?

Reply:
I'm not sure it warrants a warning, because there might not always be a sample in each section of the sampling grid for every segment type.

Here's types and locations in the 3x3 grid where no paths are found on the 50T:
```
BRAM_CASCADE(1, 0-2)
BRAM_IMUX(1, 0-2)
BUFG_CASCADE(0-1, 0-2)
CCIO_CLK_IN(1, 0-2)
CCIO_CLK_IN(2, 0)
CCIO_OUTPINFEED(1, 0-2)
CCIO_OUTPINFEED(2, 0)
GCLK(0-1, 0-2)
GCLK_INPINFEED(0-1, 0-2)
GCLK_INPINFEED(2, 0-1)
GCLK_OUTPINFEED(0-1, 0-2)
GCLK_OUTPINFEED(2, 0-1)
GND_NET(0, 0)
GND_NET(0, 2)
GND_NET(2, 0)
HCLK_CK_IN(1, 0-2)
HCLK_ROWS(0-1, 0-2)
PLL_INPINFEED(1, 0-2)
PLL_INPINFEED(2, 0)
PLL_OUTPINFEED(1, 0-2)
PLL_OUTPINFEED(2, 0)
VCC_NET(0, 0)
VCC_NET(0, 2)
VCC_NET(2, 0)
unknown(1-2, 0)
```

Reply:
Changed to `VTR_LOG_WARN`